src/garage/torch/modules/cnn_module.py

```diff
@@ -0,0 +1,276 @@
+"""CNN Module."""
+import copy
+
+from torch import nn
+
+
+# pylint: disable=unused-argument
+class CNNModule(nn.Module):
+    """Convolutional neural network (CNN) model in pytorch.
+
+    Note:
+        Based on 'NCHW' data format: [batch_size, channel, height, width].
+
+    Args:
+        input_var (int): Dimension of the network input.
+        kernel_sizes (tuple[int]): Dimension of the conv filters.
+            For example, (3, 5) means there are two convolutional layers.
+            The filter for first layer is of dimension (3 x 3)
+            and the second one is of dimension (5 x 5).
+        strides (tuple[int]): The stride of the sliding window. For example,
+            (1, 2) means there are two convolutional layers. The stride of the
+            filter for first layer is 1 and that of the second layer is 2.
+        hidden_channels (tuple[int]): Number of output channels for CNN.
+            For example, (3, 32) means there are two convolutional layers.
+            The filter for the first conv layer outputs 3 channels
+            and the second one outputs 32 channels.
+        hidden_nonlinearity (callable or torch.nn.Module):
+            Activation function for intermediate dense layer(s).
+            It should return a torch.Tensor. Set it to None to maintain a
+            linear activation.
+        hidden_w_init (callable): Initializer function for the weight
+            of intermediate dense layer(s). The function should return a
+            torch.Tensor.
+        hidden_b_init (callable): Initializer function for the bias
+            of intermediate dense layer(s). The function should return a
+            torch.Tensor.
+        paddings (tuple[int]):  Zero-padding added to both sides of the input
+        padding_mode (str): The type of padding algorithm to use,
+            either 'SAME' or 'VALID'.
+        max_pool (bool): Bool for using max-pooling or not.
+        pool_shapes (tuple[int]): Dimension of the pooling layer(s). For
```

Comment:
Reading from the implementation, I think this argument should be renamed to `pool_shape` - You should mention that all pooling layers are of the same shape.